// Decompiled with JetBrains decompiler
// Type: Coache.Core.SqlExceptionHandler
// Assembly: IsCopyConstruc, Version=5.6.2.1, Culture=neutral, PublicKeyToken=null
// MVID: 1DA50973-FF9A-4563-8B8F-74E2573677DF
// Assembly location: C:\Users\Administrateur\Downloads\Bazaar.2022.02-msil\HEUR-Trojan-Ransom.MSIL.Blocker.gen-136935297fa08738f6957068951067c699c91b2902f3ea14da650289cabfd29d.exe

using System.Collections.Generic;
using System.Data.SqlClient;
using System.Windows.Forms;

namespace Coache.Core
{
  internal class SqlExceptionHandler
  {
    public static void Handle(SqlException exception)
    {
      string upperInvariant = exception.Message.ToUpperInvariant();
      foreach (string key in SqlExceptionHandler.ConstraintMessages.Keys)
      {
        if (upperInvariant.Contains(key))
        {
          int num = (int) MessageBox.Show(SqlExceptionHandler.ConstraintMessages[key]);
          return;
        }
      }
      int num1 = (int) MessageBox.Show(exception.Message);
    }

    private static Dictionary<string, string> ConstraintMessages => new Dictionary<string, string>()
    {
      {
        "IX_USUARIO",
        "El nombre de usuario ya existe, seleccione uno nuevo"
      },
      {
        "IX_DETALLEENTIDAD_TELEFONO",
        "Ya hay un usuario con el mismo teléfono"
      },
      {
        "IX_PROVEEDOR_CUIT",
        "Ya hay un proveedor con el CUIT especificado"
      },
      {
        "IX_PROVEEDOR_RSOCIAL",
        "Ya hay un proveedor con esa Razón Social"
      },
      {
        "CK_CLIENTE_SALDO",
        "No posee saldo suficiente para realizar la compra"
      }
    };
  }
}
